doc/how-to-source-packages.rst

```diff
@@ -0,0 +1,65 @@
+How to source packages
+======================
+
+Package Source
+--------------
+
+When we publish a package, its source is available from the package repository
+[#]_. The other way to access a package's source is from a source repository [#]_.
```

Comment:
A higher level cabal-independent wording could help with context and clarity here. Probably something along the lines of
"When we publish a package, we can help users of our package to locate its source code by adding
package repository metadata [#]_ to our package. However, when we want use a package that is not, or not in the right version, available on Hackage, we can add that package that directly from its source repository (e.g. Git) [#]_."

Reply:
I'll have another go at this. I wanted to mention the two ways of getting source as a package consumer; 1) from hackage and 2) from a git repo.

![image](https://github.com/haskell/cabal/assets/633283/32d17f2b-465a-433e-98df-4928e37622d7)

Reply:
@malteneuss I've expand the introduction and added a section "Manually Obtaining Package *Source Code*" before talking about how cabal does this automatically.